Cleanup of Cellular Automaton seeding code

makortel opened the pull request:
This PR cleans up unused pieces of the CA hit triplet/quadruplet finding code (the duplication of code between triplet and quadruplet is still left there as we'll try to unify them to a more generic ntuplet finder). Essentially this means the support towards the "old seeding framework", simplifying the code to serve a clearer basis for further development.

Tested in 9_3_0_pre3, no changes expected.

@rovere @VinInn @felicepantaleo @JanFSchulte @ebrondol

makortel committed:
Remove unused pieces of CAHit{Quadruplet,Triplet}Generator

This means also removal of support for old-style seeding framework,
but I'm not sure that interface would have worked anyway nowadays
because of earlier changes.

makortel committed:
Add missing header

cmsbuild commented:
A new Pull Request was created by  @makortel (Matti Kortelainen) for master.

It involves the following packages:

RecoPixelVertexing/PixelTriplets


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@felicepantaleo, @GiacomoSguazzoni, @rovere, @VinInn, @ebrondol, @dgulhan this is something you requested to watch as well.
@davidlange6, @slava77 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

makortel commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20177/133

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-20177/133/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-20177/133/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

makortel commented:
What is the logic to select the files to be given to the code style checker?

cmsbuild commented:
+1
Tested at: 8eb42dfe14c4bc77dad4dc51b8fc8d4f36b06bc1
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20177/22317/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/7d6f190a9a22b134851a160272dc06c46c1a5063
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20177/22317/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20177/22317/git-merge-result

cmsbuild commented:
Comparison job queued.

rovere commented:
Maybe the files modified by the PR?

On Tue, Aug 15, 2017, 17:01 Matti Kortelainen <notifications@github.com>
wrote:

> What is the logic to select the files to be given to the code style
> checker?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/20177#issuecomment-322493718>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABHaR9uixniQJx1yKgZE6PVBMZxUhFHRks5sYbJ5gaJpZM4O3r_6>
> .
>
-- 
Ciao,
--Marco.

___________
Marco Rovere
Marco.Rovere@cern.ch
CERN EP-CMG-CO | room 40 3-A28 | tel +41227671209 (71209)

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20177/22317/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2718193
 - DQMHistoTests: Total failures: 286
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2717718
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

makortel commented:
> Maybe the files modified by the PR?

I think I got the logic now. I got confused because the suggested patch
https://cmssdt.cern.ch/SDT/code-checks/PR-20177/133/git-diff.patch
touches mostly files I did not edit (except `CAHitNtupletEDProducerT.cc`), but of course if I touch a .cc file, the checker also checks all the included headers.

What I don't understand though is how `PixelQuadrupletMergerEDProducer.cc` and `QuadrupletSeedMerger.cc` ended up in the list of selected files
https://cmssdt.cern.ch/SDT/code-checks/PR-20177/133/changed-files-selected.log
(or that's what I assume that this is the list of files to be checked by clang-tidy).

slava77 commented on RecoPixelVertexing/PixelTriplets/plugins/CAHitQuadrupletGenerator.h:
```diff
@@ -38,8 +37,8 @@ class CAHitQuadrupletGenerator : public HitQuadrupletGenerator {
+    CAHitQuadrupletGenerator(const edm::ParameterSet& cfg, edm::ConsumesCollector&& iC): CAHitQuadrupletGenerator(cfg, iC) {}
+    CAHitQuadrupletGenerator(const edm::ParameterSet& cfg, edm::ConsumesCollector& iC);
 
     virtual ~CAHitQuadrupletGenerator();
```
is ```virtual``` still needed here?

VinInn replied:
I suppose is in https://cmssdt.cern.ch/SDT/code-checks/PR-20177/133/git-diff.patch
that in my understanding has not been applied yet...

slava77 replied:
To be more clear: CAHitQuadrupletGenerator does not derive from anything anymore and there is no derived class from it either. Why do we need a virtual destructor in this case?

slava77 commented on RecoPixelVertexing/PixelTriplets/plugins/CAHitTripletGenerator.h:
```diff
@@ -29,16 +28,16 @@ namespace edm {
+    CAHitTripletGenerator(const edm::ParameterSet& cfg, edm::ConsumesCollector&& iC): CAHitTripletGenerator(cfg, iC) {}
+    CAHitTripletGenerator(const edm::ParameterSet& cfg, edm::ConsumesCollector& iC);
 
     virtual ~CAHitTripletGenerator();
```
is ```virtual``` still needed?

makortel commented:
Thanks Slava, the virtual destructor is certainly not needed (slipped through my eyes). I'll remove it early next weeknor so.

On 21 August 2017 16:42:54 CEST, Slava Krutelyov <notifications@github.com> wrote:
>slava77 commented on this pull request.
>
>
>
>> @@ -38,8 +37,8 @@ class CAHitQuadrupletGenerator : public
>HitQuadrupletGenerator {
> 
> public:
> 
>-    CAHitQuadrupletGenerator(const edm::ParameterSet& cfg,
>edm::ConsumesCollector&& iC, bool needSeedingLayerSetsHits=true):
>CAHitQuadrupletGenerator(cfg, iC, needSeedingLayerSetsHits) {}
>-    CAHitQuadrupletGenerator(const edm::ParameterSet& cfg,
>edm::ConsumesCollector& iC, bool needSeedingLayerSetsHits=true);
>+    CAHitQuadrupletGenerator(const edm::ParameterSet& cfg,
>edm::ConsumesCollector&& iC): CAHitQuadrupletGenerator(cfg, iC) {}
>+    CAHitQuadrupletGenerator(const edm::ParameterSet& cfg,
>edm::ConsumesCollector& iC);
> 
>     virtual ~CAHitQuadrupletGenerator();
>
>To be more clear: CAHitQuadrupletGenerator does not derive from
>anything anymore and there is no derived class from it either. Why do
>we need a virtual destructor in this case?

felicepantaleo commented on RecoPixelVertexing/PixelTriplets/plugins/CAHitQuadrupletGenerator.cc:
```diff
@@ -54,18 +54,11 @@ caPhiCut(cfg.getParameter<double>("CAPhiCut")),
 caHardPtCut(cfg.getParameter<double>("CAHardPtCut")),
 caOnlyOneLastHitPerLayerFilter(cfg.getParameter<bool>("CAOnlyOneLastHitPerLayerFilter"))
```
hi @makortel , could you also get rid of the `caOnlyOneLastHitPerLayerFilter` variable in this PR?

makortel committed:
Remove virtual destructors as they became unnecessary

makortel committed:
Remove CAOnlyOneLastHitPerLayerFilter parameter as obsolete

makortel committed:
Code style checks

makortel committed:
Remove obsolete python configuration and customisation files

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #20177 was updated. @perrotta, @cmsbuild, @slava77 can you please check and sign again.

makortel commented:
@cmsbuild, please test

* removed `virtual` from `CAHit*Generator` destructors (suggested by @slava77)
* removed `CAOnlyOneLastHitPerLayerFilter` parameter (suggested by @felicepantaleo)
* applied code check patch
* removed obsolete python configuration files

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20177/341

cmsbuild commented:
+1
Tested at: 2be23e05cbb1dab188263d8fda56a2091b43eab2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20177/22518/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5ca2c40df047c88cd2090929074e535151529687
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20177/22518/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20177/22518/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20177/22518/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2653934
 - DQMHistoTests: Total failures: 203
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2653542
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/20177	2be23e0
- cleanup of obsolete CA code; clang tidy
- jenkins tests pass and comparisons show no differences

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

davidlange6 commented:
+1
